src/foam/nanos/menu/DAOMenu.js

```diff
@@ -133,9 +142,14 @@ foam.CLASS({
         throw new Error('No DAO found for ' + this.daoKey);
       }
 
+      var dao = X[this.daoKey];
+      dao.of = this.of ? this.of : dao.of;
```

Comment:
Wouldn't this be changing the `of` for that DAO everywhere? For example, if you set `of` to Business and `daoKey = "userDAO"`, then you'd be setting the `of` of `userDAO` to `Business` on your client which might affect other parts of the client code that expect it to be `User`. I'm not sure if this assignment will affect that wide of a scope or not though. I'd have to test to see.

Reply:
Yes we shouldn't mutate the `of` property of the dao.

If you want to make sure the DAOController only creates business objects, what you could do is create a new ProxyDAO which has it's own `of` set to `Business` and the delegate of which is a predicated dao for `INSTANCE_OF(Business)`

Reply:
Couldn't we clone the DAO and set the of in the DAOMenu? That would prevent the DAO within the context to be mutated.